### PR TITLE
Don't use watcher function for logging job status

### PR DESCRIPTION
**Pull request by rabi**
This also optimizes to only query failed pods when the deployment has failed.

### CODE REVIEW

**Comment by rabi** [comment]
Last failure `Failed to run 'make crc_storage' target. Aborting`

File: controllers/dataplane/openstackdataplanenodeset_controller.go [hunk 1]
```diff
@@ -436,31 +436,38 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
...
+		if _err != nil {
+			Log.Error(err, "unable to retrieve list of pods for dataplane diagnostic")
+		} else {
+			for _, pod := range podsList.Items {
+				Log.Info(fmt.Sprintf("openstackansibleee job %s failed due to %s with message: %s", pod.Name, pod.Status.Reason, pod.Status.Message))
+			}
 		}
-		instance.Status.Conditions.MarkFalse(condition.DeploymentReadyCondition,
-			condition.ErrorReason, condition.SeverityError,
-			deployErrorMsg)
```
Comment (fao89) [inline controllers/dataplane/openstackdataplanenodeset_controller.go:446]: I think we should keep the status update

Comment (rabi) [reply controllers/dataplane/openstackdataplanenodeset_controller.go:446]: oops that was unintentional.

**Pull request merged**